src/std/rec/waveformRecord.dbd.pod

```diff
@@ -1,45 +1,508 @@
+types.
 
-=head2 Record-specific Menus
+=recordtype waveform
+
+=cut
+
+menu(waveformPOST) {
+    choice(waveformPOST_Always,"Always")
+    choice(waveformPOST_OnChange,"On Change")
+}
+
+recordtype(waveform) {
 
-=head3 Menu waveformPOST
+=pod
 
-The MPST and APST fields use this menu to determine when to post new value
-and archive monitors respectively.
+=head1 Contents
 
-=menu waveformPOST
+=over
+
+=item * L<Parameter Fields>
+
+=over
+
+=item * L<Scan Parameters>
+
+=item * L<Read Parameters>
+
+=item * L<Operator Display Parameters>
+
+=item * L<Run-time Parameters>
+
+=back
+
+=item * L<Record Support>
+
+=over
+
+=item * L<Record Support Routines>
+
+=item * L<Record Processing>
+
+=back
 
-...
+=item * L<Device Support>
+
+=over
+
+=item * L<Fields Of Interest To Device Support>
+
+=item * L<Device Support Routines>
+
+=item * L<Device Support For Soft Records>
+
+=back
+
+=back
+
+=begin html
+
+<hr>
+
+=end html
 
 =head2 Parameter Fields
 
-The record-specific fields are described below.
+The waveform's fields fall into the following categories:
 
-=recordtype waveform
+=over
 
-...
+=item * L<Scan Parameters>;
 
-=cut
+=item * L<Read Parameters>;
 
-menu(waveformPOST) {
-    choice(waveformPOST_Always,"Always")
-    choice(waveformPOST_OnChange,"On Change")
-}
+=item * L<Operator Display Parameters>;
 
-recordtype(waveform) {
+=item * L<Run-time Parameters>;
+
+=back
+
+=head3 Scan Parameters
+
+The waveform record has the standard fields for specifying under what
+circumstances the record will be processed. These fields are listed in L<Scan
+Fields>. In addition, L<Scanning Specification> explains how these fields are
+used. Note that I/O event scanning is only supported for those card types that
+interrupt.
+
+=head3 Read Parameters
+
+These fields are configurable by the user to specify how and from where the
+record reads its data. How the INP field is configured determines where the
+waveform gets its input. It can be a hardware address, a channel access or
+database link, or a constant. Only in records that use soft device support can
+the INP field be a channel access link, a database link, or a constant.
+Otherwise, the INP field must be a hardware address. See L<Address
+Specification> for information on the format of hardware addresses and database
+links.
+
+=head4 Fields related to waveform reading
+
+=fields DTYP, INP, NELM, FTVL, RARM
+
+The DTYP field must contain the name of the appropriate device support module.
+The values retrieved from the input link are placed in an array referenced by
+VAL. (If the INP link is a constant, elements can be placed in the array via
+dbPuts.) NELM specifies the number of elements that the array will hold, while
+FTVL specifies the data type of the elements.
+The RARM field causes the device to re-arm when this field is set to 1. 
+
+=head4 Possible data types for FTVL
+
+=begin html
+
+<blockquote><table border="1"><tr>
```

Comment:
Is this table just a copy of the output from the `=menu menuFtvl` POD conversion? There's a better way to include that by actually including the menu file into the DBD part, then it's available for the `=menu` keyword:
```patch
--- a/src/std/rec/waveformRecord.dbd.pod
+++ b/src/std/rec/waveformRecord.dbd.pod
@@ -22,6 +22,10 @@ and archive monitors respectively.
 
 ...
 
+=head3 Menu FTVL
+
+=menu menuFtype
+
 =head2 Parameter Fields
 
 The record-specific fields are described below.
@@ -32,6 +36,8 @@ The record-specific fields are described below.
 
 =cut
 
+include "menuFtype.dbd"
+
 menu(waveformPOST) {
     choice(waveformPOST_Always,"Always")
     choice(waveformPOST_OnChange,"On Change")
```

If the waveformRecord.c file had a `#include "menuFtype.h"` it would be necessary to delete that include line to avoid the C compiler seeing duplicate definitions (in this case it doesn't, but other records might) but this approach does work and is better.

Reply:
Thanks for the hint again, this is exactly what I wanted to do but didn't know how.